### PR TITLE
Update stk to Trilinos

### PR DESCRIPTION
**Pull request by alanw0**
Bring in ngp::Field versions of parallel_sum, copy_owned_to_shared, and
communicate_field_data.

Also fix some openmp compile issues reported by Roger Pawlowski.

@trilinos/<teamName>

## Checklist

- [ ] My commit messages mention the appropriate GitHub issue numbers.
- [ ] My code follows the code style of the affected package(s).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [code contribution guidelines](../blob/master/CONTRIBUTING.md) for this project.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] No new compiler warnings were introduced.
- [ ] These changes break backwards compatibility.

### CODE REVIEW

File: packages/stk/stk_ngp/stk_ngp/NgpField.hpp [hunk 1]
```diff
@@ -379,6 +379,14 @@ class StaticField : public FieldBase
 
     const stk::mesh::BulkData& get_bulk() const { return *hostBulk; }
 
+    STK_FUNCTION
+    void swap_data(StaticField<T> &sf)
```
Comment (mhoemmen) [inline packages/stk/stk_ngp/stk_ngp/NgpField.hpp]: Why not just call this "swap"?

Comment (alanw0) [reply packages/stk/stk_ngp/stk_ngp/NgpField.hpp]: Yes, swap would be just as good here. Maybe there's even a better name. The intent is to do a "shallow" swap. I.e., in traditional sierra-framework fields it was just a pointer swap, not a copying of the data. It is for managing multi-state fields, e.g., an 'n' and 'n+1' state.

Comment (mhoemmen) [reply packages/stk/stk_ngp/stk_ngp/NgpField.hpp]: @alanw0 It looks like this means the same thing as, say, `std::shared_ptr` swap:

https://en.cppreference.com/w/cpp/memory/shared_ptr/swap

If so, then I think `swap` is the right name for this function.

Comment (alanw0) [reply packages/stk/stk_ngp/stk_ngp/NgpField.hpp]: done.

**Pull request merged**